### PR TITLE
Assitant Input Bar implementation

### PR DESCRIPTION
**Pull request by spolu**
cc @dust-tt/engineering-team 

This is 98% there. There is one quite edge-casy bug preventing deleting a mention if it is at the beginning of the input bar by doing backspace (ctrl-a + delete do work). Only happens after you created and deleted a bunch of mentions and recreate one at the beginning.

Basically we manage a content editable from the InputBar component and rely a list of mentions component that we show and hide on demand close to the cursor. That component exposes an API (forwardRef + useImpearativeHandle machinery) to drive the selection (up / down, enter).

It is arguable that we could drive that state from InputBar instead with an onHover(focusIndex) on the AgentList. I'm not 100% convinced it's possible but could be explored, I went this way because I knew I could get to it with that approach.

### CODE REVIEW

**Comment by spolu** [comment]
I would suggest checking it out and playing with it as a starting point.

Note: we couldn't use the Doprdown components for the AgentList (we basically clone their design which will require syncrhonization as sparkle evolve) for a couple reason:
- It has to be moved where the cursor
- we can't inject even a hidden dropdown button in the contenteditable as we're off react path, we can just inject html there

**Comment by spolu** [comment]
I also looked at react-mentions but we couldn't have achieved the same result (they overlay spans over an input and you can't recover the input text so you really can only add color background to it and not change the style. Very unclear how we would have achieved the design with avatars in the list of agents)

**Review by philipperolet — APPROVED** [review]
Impressive and beautiful result :star_struck: 

Tested it on desktop / mobile => works great
I had a bit trouble reading the code, tough to get but what I understand is :+1: 
Little nit comment

**Review by fontanierh — APPROVED** [review]

**Review by PopDaph — APPROVED** [review]
A few remaining console.log but looks great 👍🏻

File: front/components/assistant/conversation/InputBar.tsx [hunk 1]
```diff
@@ -1,74 +1,575 @@
...
+            <div className={classNames("z-10 flex flex-row space-x-4 pr-2")}>
+              <div className="flex flex-col justify-center">
+                <DropdownMenu>
+                  <DropdownMenu.Button icon={RobotIcon} />
+                  <DropdownMenu.Items origin="bottomRight">
+                    {agentConfigurations.map((c) => (
+                      <DropdownMenu.Item
+                        key={c.sId}
+                        label={"@" + c.name}
+                        visual={c.pictureUrl}
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant/conversation/InputBar.tsx:522]: It does on 0.1.66

[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant/conversation/InputBar.tsx:522]: On the DropdownMenu.Item otherwise we wouldn't see the avatar?

**Pull request merged**